source/agora/app.d

```diff
@@ -12,12 +20,38 @@
 *******************************************************************************/
 
 module agora.app;
+import agora.common.Config;
+import agora.node.Node;
+
+import vibe.core.args;
+import vibe.core.core;
+import vibe.core.log;
+import vibe.http.router;
+import vibe.http.server;
+import vibe.web.rest;
 
-void main()
+import std.digest.sha;
+import std.stdio;
+import core.stdc.time;
+import core.time;
+
+/// todo: work around this for 'dub test'
+version (unittest) void main () { } else:
+
+/// Application entry point
+public int main (string[] args)
 {
-	// TODO:
-    // - Config parsing
-    // - Network interface setup
-    // - Event loop
-    // - End world hunger
+    CommandLine cmdln;
+    if (!parseCommandLine(cmdln, args))
+        return 0;  // help shown, early exit
```

Comment:
```suggestion
        return 1;  // help shown, early exit
```

Reply:
^ After reading the code, forget about this

Reply:
I could make it an enum to make it clear?

Reply:
Some command line tools returns success on `--help` (e.g. `git`), some don't.
Let's follow git here